_source/_posts/2022-09-09-supabase-vue.md

```diff
@@ -0,0 +1,941 @@
+
+> description: text
+> isPublic: bool
+> date: timestamptz
+> title: text
+
+Make each column non-nullable.  By default, Supabase makes all columns nullable. Then click **Save**. After a moment you will see your table in the _Table editor_ section.
+
+{% img blog/supabase-vue/supabase-create-table.png alt:"Create a table in supabase" width:"600" %}{: .center-image }
+
+### Populate some sample data
+
+Click on the table you just created in the table editor and click **Insert Row**. This will bring up an editor where you can add new data. I added 3 entries: two public and one private entry. Give your entries dates that are in your current month. For example, I'm writing this article in September 2022 so I used three dates in that month.
+
+{% img blog/supabase-vue/supabase-add-row.png alt:"Add data to supabase" width:"600" %}{: .center-image }
+
+After each row, click save and you should see the data you complete populated in the Table editor.
+
+## Create your Vue Application
+
+It's now time to create your Vue application. Start by running the following command.
+
+```sh
+npm init vue@3.3.2 OktaSupabase
+```
+
+This will create a new Vue application called `OktaSupabase` in a subfolder on your working directory with the same name. There will be several options in this command such as the project name, TypeScript support, JSX support, and others. See the image below for the options I've chosen. If you want to use some of these options feel free to turn them on.
+
+{% img blog/supabase-vue/vue-init-options.png alt:"Create your vue application" width:"600" %}{: .center-image }
+
+### Install your dependencies
+
+Now you can install the dependencies you need for this project.
+
+> > Tanay : trying something a little different here. If its no good let me know.
+
+#### @okta/okta-vue
+
+```sh
+npm i @okta/okta-vue@5.3.0
+```
+
+The [Okta Vue SDK](https://github.com/okta/okta-vue) makes authenticating and securing your application with Okta quick and easy.
+
+#### @supabase/supabase-js
+
+```sh
+npm i @supabase/supabase-js@1.35.6
+```
+
+The [Supabase JavaScript client](https://github.com/supabase/supabase-js) provides all the functionality you will need to connect to your Supabase instance and interact with your database.
+
+#### daisyui
+
+```sh
+npm i daisyui@2.24.0
+```
+
+[DaisyUI](https://daisyui.com/) is a tailwind component library that will help make your design work a little easier.
+
+#### daisyUI
+
+```sh
+npm i vue3-datepicker@0.3.4
+```
+
+[vue3-datepicker](https://github.com/icehaunter/vue3-datepicker) is a datepicker for Vue3.
+
+### Add .env
+
+Next, add a file called `.env` to your root directory with the following code.
+
+```
+VITE_SUPABASE_URL={yourSupabaseUrl}
+VITE_SUPABASE_ANON_KEY={yourSupabaseAnonKey}
+VITE_OKTA_DOMAIN={yourOktaDomain}
+VITE_OKTA_CLIENTID={yourClientID}
+```
+
+These values are populated from the Okta CLI and the Supabase web interface.
```

Comment:
The Okta CLI would create a `.okta.env` file, right? Should the users copy the value from that and add it to this `.env`? Can they then delete the `.okta.env` file?